hmf/density_field/transfer_models.py

```diff
@@ -713,3 +718,65 @@ class EH(EH_BAO):
     """Alias of :class:`EH_BAO`."""
 
     pass
+
+
+class CLASS(FromFile):
+    """
+    Transfer function computed by CLASS.
+
+    Parameters
+    ----------
+
+    cosmo : :class:`astropy.cosmology.FLRW` instance
+        The cosmology used in the calculation
+
+    \*\*model_parameters : unpack-dict
+        Parameters specific to this model.
+
+        **class_params:**
+            Dict of params to set for CLASS object
+        **class_object:**
+            Custom class object. If not set, a new Class object will be
+            created.
+        **flip_t:**
+            If set to True, will return ln(|T|).
+    """
+    _defaults = {"class_params": None, "class_obj": None,
+                 "flip_T": False}
+
+    def __init__(self, *args, **kwargs):
+        super(CLASS, self).__init__(*args, **kwargs)
+        self.spline_fn = None
+
+        if self.params['class_obj'] is not None:
+            self.class_obj = self.params['class_obj']
+        else:
+            h = self.cosmo.h
+            class_params = {
+                'omega_b': self.cosmo.Ob0 * h ** 2,
+                'omega_cdm': (self.cosmo.Om0 - self.cosmo.Ob0) * h ** 2,
+                'h': h,
+            }
+            if self.params['class_params'] is not None:
+                class_params.update(self.params['class_params'])
+            class_params['output'] = 'dTk'
+
+            self.class_obj = classy.Class()
+            self.class_obj.set(class_params)
+
+    def lnt(self, lnk):
+        if self.spline_fn is None:
+            trans = self.class_obj.get_transfer(output_format='camb')
+            if not trans:
+                self.class_obj.compute()
+                trans = self.class_obj.get_transfer(output_format='camb')
```

Comment:
I'm not very familiar with Class, but I feel like there might be a more elegant way to do this. Is there no other way to check if `compute` has been called other than to get the transfer and check if it's not None? Ideally, you'd have:

```
if not self.class_obj.condition:
    self.class_obj.compute()

trans = self.class_obj.get_transfer(output_format='camb')